### DISCUSSION TITLE
怎么获取MSVC版本或者GCC版本呢

### DISCUSSION THREAD

**Discussion by ipyffor** [xmake-io/xmake#1855] [Q&A]
> Note: If it is a 'bug report' or 'feature request', please select the corresponding issue template, otherwise this issue will be not replied. 

### Describe the problem details
Please describe your problem details here.
- 怎么获取MSVC版本或者GCC版本呢
- linux下怎么通过find_packages找到cuda
- 通过find_packages找到的库能指定链接部分库而不是添加所有的库吗
- print打印的字符串会会输出多次问题

**Comment by xq114** [comment]
> * 怎么获取MSVC版本或者GCC版本呢

local vs = import("core.tool.toolchain").load("msvc"):config("vs")

> * linux下怎么通过find_packages找到cuda

add_requires("cuda", {system=true, configs={utils={"cublas", "cufft", ...}}})

> * 通过find_packages找到的库能指定链接部分库而不是添加所有的库吗

目前不能，一个package的库是固定的，后续改进见 https://github.com/xmake-io/xmake/issues/1452

> * print打印的字符串会会输出多次问题

xmake有描述域和脚本域的概念，on_config/on_load/before_build等函数里面才是脚本逻辑，写在外面的都是描述文件，默认并发执行，并且会多次解释，所以不要把print写在函数外面的描述域

**Comment by waruqi** [comment]
其他的上面都说的很清楚了。

> print打印的字符串会会输出多次问题

看下文档，有详细描述

https://xmake.io/#/zh-cn/guide/syntax_description?id=%e9%85%8d%e7%bd%ae%e5%88%86%e7%a6%bb
